parse.go

```diff
@@ -61,114 +59,180 @@ type Template struct {
 }
 type Handle uintptr
 
+var debug = false
+var mw1 *walk.MainWindow
+var configButton, debugButton, closedButton *walk.PushButton
+var imgView *walk.ImageView
+var debugTxt *walk.TextEdit
+var debugGrpBox *walk.GroupBox
+var cafilePath string
+
+var tempPath string
+var pngFileName string
+var pngFilePath string
+var userCertPath string
+var profilePath string
+var templateOutPath string
+var stableCurrentWorkingDirectory string
+
 func main() {
 	hideConsole()
+	prepareMainWindow()
+	if err := (MainWindow{
+		AssignTo:   &mw1,
+		Title:      fmt.Sprintf("%s - %s", PROGRAM_NAME, VERSION),
+		MinSize:    Size{Width: 500, Height: 400},
+		Layout:     VBox{},
+		Background: SolidColorBrush{Color: walk.RGB(4, 5, 3)},
+		Children: []Widget{
+			ImageView{
+				AssignTo:   &imgView,
+				Background: SolidColorBrush{Color: walk.RGB(4, 5, 3)},
+				//Image:      pngFileName,
+				Visible: true,
+			},
+			GroupBox{
+				Title:    "Configuration Debug output",
+				AssignTo: &debugGrpBox,
+				Visible:  false,
+				Layout:   VBox{},
+				Children: []Widget{
+					ScrollView{
+						HorizontalFixed: true,
+						Layout:          VBox{},
+						Children: []Widget{
+							TextEdit{
+								MinSize:  Size{Width: 400, Height: 10},
+								AssignTo: &debugTxt,
+								ReadOnly: false,
+								Text:     "",
+							},
+						},
+					},
+				},
+			},
+			PushButton{
+				AssignTo:   &configButton,
+				Background: SolidColorBrush{Color: walk.RGB(4, 5, 3)},
+				MinSize:    Size{Width: 50, Height: 50},
+				Text:       "Configure",
+				OnClicked: func() {
+					fetchPortalDomainName()
+				},
+			},
+			PushButton{
+				AssignTo:   &debugButton,
+				Background: SolidColorBrush{Color: walk.RGB(4, 5, 3)},
+				MinSize:    Size{Width: 10, Height: 10},
+				Text:       "Debug",
+				OnClicked: func() {
+					viewDebug()
+				},
+			},
+			PushButton{
+				AssignTo:   &closedButton,
+				Background: SolidColorBrush{Color: walk.RGB(4, 5, 3)},
+				MinSize:    Size{Width: 10, Height: 10},
+				Text:       "Close",
+				Visible:    false,
+				OnClicked: func() {
+					cleanAndExit(0)
+				},
+			},
+		},
+		OnSizeChanged: func() {
+			mw1size := Size{Width: 500, Height: 400}
+			mw1.SetSize(walk.Size(mw1size))
+		},
+	}.Create()); err != nil {
+		log.Print("Failed opening main window: ", err.Error())
+		viewErrorAndExit(T("errorMainWindow: ")+""+err.Error(), "")
+	}
+	prepareEnv()
+	prepareBackgroundImage()
 
-	log.Println("==================== PacketFence Provisioning Agent ===================")
+	mw1.Run()
+}
 
+func prepareMainWindow() {
 	currentWorkingDirectory, err := os.Executable()
 	if err != nil {
 		walk.MsgBox(windowMsgBox, "Error", "Unable to get current working directory, please contact your local support.", walk.MsgBoxOK)
```

Comment:
Well, this message box will stop the program before the main window is open. It should be good to keep it in case we do not have access to this directory, and every files created during the process are based on this path.